R/EventStudy.R

```diff
@@ -170,56 +174,90 @@ EventStudy <- function(estimator, data, outcomevar, policyvar, idvar, timevar, c
+            warning(paste0("Variables starting with ", policyvar, tag,
+                           " should be reserved for usage by eventstudyr."))
+        }
+    }
 
-    num_fd_lag_periods   <- post + overidpost - 1
-    num_fd_lead_periods  <- pre + overidpre
+    # Compute shifts in policy variable
+    num_fd_lags  <- post + overidpost - 1
+    num_fd_leads <- pre  + overidpre
 
-    furthest_lag_period  <- num_fd_lag_periods + 1
+    furthest_lag_period <- num_fd_lags + 1
 
-    if (post + overidpost + pre + overidpre > 0) {
-        data <- GetFirstDifferences(df = data, groupvar = idvar, timevar, diffvar = policyvar)
-    }
+    if (static) {
+        message("post, overidpost, pre, and overidpre are set to 0. A static model will be estimated.")
+    } else {
+        data <- ComputeFirstDifferences(data, idvar, timevar, policyvar, timevar_holes)
 
-    if (post + overidpost - 1 >= 1) {
-        data <- PrepareLags(data, groupvar = idvar, timevar,
-                            lagvar = paste0(policyvar, "_fd"), lags = 1:num_fd_lag_periods)
-    }
+        if ((post + overidpost - 1 >= 1) & (pre + overidpre >= 1)) {
+            shift_values = c(-num_fd_leads:-1, 1:num_fd_lags)
+        }
+        else if (pre + overidpre < 1) {
+            shift_values = 1:num_fd_lags
+        }
+        else if (post + overidpost - 1 < 1) {
+            shift_values = -num_fd_leads:-1
+        }
 
-    if (pre + overidpre >= 1) {
-        data <- PrepareLeads(data, groupvar = idvar, timevar, leadvar = paste0(policyvar, "_fd"),
-                             leads = 1:num_fd_lead_periods)
+        data <- ComputeShifts(data, idvar, timevar,
+                              shiftvar    = paste0(policyvar, "_fd"),
+                              shiftvalues = shift_values,
+                              timevar_holes = timevar_holes)
     }
 
-    if (post == 0 & overidpost == 0 & pre == 0 & overidpre == 0) {
-        data      <- PrepareLeads(data, groupvar = idvar, timevar,
-                                  leadvar = policyvar, leads = num_fd_lead_periods)
-    } else {
-        data <- PrepareLags(data, groupvar = idvar, timevar,
-                            lagvar = policyvar, lags = furthest_lag_period)
-        data <- PrepareLeads(data, groupvar = idvar, timevar,
-                             leadvar = policyvar, leads = num_fd_lead_periods)
+    if (!static) {
+        data <- ComputeShifts(data, idvar, timevar,
+                              shiftvar    = policyvar,
+                              shiftvalues = c(-num_fd_leads, furthest_lag_period),
+                              timevar_holes = timevar_holes)
 
-        column_subtract_1 <- paste0(policyvar, "_lead", num_fd_lead_periods)
-        data[column_subtract_1] <- 1 - data[column_subtract_1]
+        lead_endpoint_var <- paste0(policyvar, "_lead", num_fd_leads)
+        data[lead_endpoint_var] <- 1 - data[lead_endpoint_var]
```

Comment:
@santiagohermo why are we performing this subtraction here?

Reply:
If you look at the equation in the paper you'll see this term! It turns out the `1-` is not needed, but we keep it so that the implementation is aligned with the paper. I asked about this around [here](https://github.com/JMSLab/eventstudyr/issues/31#issuecomment-1531558311)